pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -533,6 +519,16 @@ func (o *MustGatherOptions) Run() error {
 func (o *MustGatherOptions) podProcessor(ns string, podChan chan *corev1.Pod, errCh chan error, wg *sync.WaitGroup) {
 	var err error
 	for pod := range podChan {
+		_, err = o.Client.CoreV1().Pods(ns).Create(context.TODO(), pod, metav1.CreateOptions{})
+		if err != nil {
+			errCh <- err
+			return
+		}
```

Comment:
@ingvagabund in "master" branch, we exit immediately if creation of any pod fails. In above code, we return from the goroutine where pod creation fails. I don't think it's accurate because other goroutines would still keep running.

One thing I could think of is to use [`errgroup.Group`](https://pkg.go.dev/golang.org/x/sync/errgroup#Group) which will error out if any of the goroutines returns an error.

Another could be to use a `select` statement that would take over the control and stop goroutines (using `context.Cancel`.

What do you think?

Reply:
> we return from the goroutine where pod creation fails. I don't think it's accurate because other goroutines would still keep running.

Other goroutines can continue working and we can return the aggregated error at the end. Otherwise, if there is a problematic image, that directly affects the other must-gather images processes.

Reply:
> Other goroutines can continue working and we can return the aggregated error at the end

OK. In that case, I would prefer we don't `return` and simply put the `err` on `errCh`. So that we still have `concurrentMG` number of goroutines running. Basically, just remove `return` from the code I have highlighted.